lib/mathlib.cpp

```diff
@@ -548,26 +548,54 @@ MathLib::bigint MathLib::toLongNumber(const std::string & str)
     return ret;
 }
 
-// Assuming a limited support of built-in hexadecimal floats (see C99, C++17) that is a fall-back implementation.
-// For now it's not optimized WRT performance at all...
-static double FloatHexToDoubleNumber(const std::string& str)
+// in-place conversion of (sub)string to double. Requires now heap.
```

Comment:
A typo... should be "no heap" (or at least that's what I wanted to achieve).
Certainly dropping that requirement would allow to omit that function.